src/main/resources/Diagram/DiagramEditSheet.xml

```diff
@@ -767,6 +774,24 @@ define('diagramStore', ['jquery', 'xwiki-meta', 'xwiki-utils', 'draw.io', 'xwiki
     oldImportFiles.apply(this, newArgs);
   }
 
+  var originalLoadUrl = EditorUi.prototype.loadUrl;
+  EditorUi.prototype.loadUrl = function(url, success, error, forceBinary, retry, dataUriPrefix) {
+    let proxyIndex = url.indexOf(PROXY_URL);
+    // Use indexOf since URLSearchParams is not supported by IE.
+    let urlParamIndex = url.indexOf('url=');
+    if (proxyIndex &gt; -1 &amp;&amp; urlParamIndex &gt; -1) {
+      // Get the value of url parameter
+      let originalUrl = unescape(url.substring(urlParamIndex+4));
```

Comment:
You're making 2 assumptions here:

* there's no other parameter in the query string after the ``url`` parameter
* there's no document fragment identifier at the end of the URL

It may be the case but it's not clean and it makes the code fragile.

Also, on https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/unescape I see this note:

> Note: Do not use unescape to decode URIs, use decodeURI instead.

Finally, I don't understand why you need to decode twice the value of the ``url`` parameter. Calling **only** ``decodeURIComponent`` should be enough I think.